app/src/main/javascript/bundles/experiment-page/src/TSnePlotViewRoute.js

```diff
@@ -53,7 +54,7 @@ class TSnePlotViewRoute extends React.Component {
       selectedColourBy: cellTypeValue ? cellTypeValue.toLowerCase() : this.props.ks[Math.round((this.props.ks.length -1) / 2)].toString(),
       highlightClusters: [],
       experimentAccession: this.props.experimentAccession,
-      selectedColourByCategory: cellTypeValue ? `metadata` : `clusters`
+      selectedColourByCategory: !isNaN(search.colourBy) ? `clusters` : cellTypeValue ? `metadata` : `clusters`
```

Comment:
Maybe `Number.isSafeInteger(search.colourBy)` is more readable? As usual there’s more than one way to do this and you decide, it’s a matter of taste really.

Reply:
I believe we will need to add an extra function to the suggested solution as `search.colourBy` is a string and for `Number.isSafeInteger` it returns `false` for anything which is not a number value. So we have two options here either we do something like this `Number.isSafeInteger(Number(search.colourBy))` to convert `search.colourBy` to a number type, and then perform `isSafeInteger` check or keep the solution as it is. I personally like the latter. Let me know what you think?

Reply:
https://www.w3schools.com/jsref/tryit.asp?filename=tryjsref_issafeinteger2

Reply:
Yes, you’re totally right. I forgot about the type coertion stuff. Leave as it is then.